cli/lsp/language_server.rs

```diff
@@ -1517,9 +1518,37 @@ impl Inner {
 
     // if the current deno.json has changed, we need to reload it
     if has_config_changed(&self.config, &changes) {
+      let mut files_to_check = HashSet::with_capacity(4);
+      if let Some(url) = self.config.maybe_config_file().map(|c| &c.specifier) {
+        files_to_check.insert(url.clone());
+      }
+      if let Some(url) = self.config.maybe_config_file_canonicalized_specifier()
+      {
+        files_to_check.insert(url.clone());
+      }
       if let Err(err) = self.update_config_file().await {
         self.client.show_message(MessageType::WARNING, err);
       }
+      if let Some(url) = self.config.maybe_config_file().map(|c| &c.specifier) {
+        files_to_check.insert(url.clone());
+      }
+      if let Some(url) = self.config.maybe_config_file_canonicalized_specifier()
+      {
+        files_to_check.insert(url.clone());
+      }
```

Comment:
Could you explain this bit? Why are we checking for both canonicalized and non canonicalized paths? Also I would appreciate comments for these - my first intuition was that this is duplicated code, only then I spotted that we're updating current config file path in between these